src/api/app/queries/notifications_finder.rb

```diff
@@ -63,39 +79,9 @@ def for_appealed_decisions
     @relation.where(event_type: 'Event::AppealCreated', delivered: false)
   end
 
-  # rubocop:disable Metrics/CyclomaticComplexity
-  # We need to refactor this method, the `case` statement is way too big
   def for_notifiable_type(type = 'unread')
-    notifications = self.class.new(with_notifiable)
-
-    case type
-    when 'read'
-      notifications.read
-    when 'comments'
-      notifications.unread.where(notifiable_type: 'Comment')
-    when 'requests'
-      notifications.unread.where(notifiable_type: 'BsRequest')
-    when 'incoming_requests'
-      notifications.for_incoming_requests
-    when 'outgoing_requests'
-      notifications.for_outgoing_requests
-    when 'relationships_created'
-      notifications.for_relationships_created
-    when 'relationships_deleted'
-      notifications.for_relationships_deleted
-    when 'build_failures'
-      notifications.for_failed_builds
-    when 'reports'
-      notifications.for_reports
-    when 'workflow_runs'
-      notifications.for_workflow_runs
-    when 'appealed_decisions'
-      notifications.for_appealed_decisions
-    else
-      notifications.unread
-    end
-  end
-  # rubocop:enable Metrics/CyclomaticComplexity
+    NOTIFIABLE_TYPE_MAP[type].call(notifications)
```

Comment:
*You* find this less complex than a `case`?

Reply:
Not really. I was not convinced by how it played out, but I pushed the PR to get feedback. I'm happy to see that someone else thinks the case was not that card to read.

Reply:
Yeah we are not slave to the numbers :)